From 1c9582b8e05aaef6f6234a15f91a31b94a9d96ec Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Mon, 22 Jun 2026 14:28:48 +0530
Subject: [PATCH] Skip unused STA detach and FD re-registration in  CONNECTED
 FSM

Commented out the unused station (STA HAL instances) reset and FD re-registration
logic for active STA in the CONNECTED FSM to prevents missing the EndPoint Object
change event in sta_wlan_hal_whm during BSTA disconnect.

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>
---
 .../beerocks/slave/backhaul_manager/backhaul_manager.cpp    | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 400612401..8d85f58d1 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -610,7 +610,9 @@ bool BackhaulManager::handle_backhaul_connect()
             db->backhaul.backhaul_bssid =
                 tlvf::mac_from_string(radio_info->sta_wlan_hal->get_bssid());
 
-            if (!radio_info->sta_wlan_hal->unique_file_descriptors() &&
+            //This code segment is commented out to avoid missing EndPoint Object change event, when
+            //BSTA disconnect
+            /*if (!radio_info->sta_wlan_hal->unique_file_descriptors() &&
                 radio_info->sta_hal_ext_events.empty()) {
                 // previous iteration destroyed the sta_wlan_hal instances that are unused
                 // if the FDs were registered with another sta_wlan_hal, re-register now
@@ -678,7 +680,7 @@ bool BackhaulManager::handle_backhaul_connect()
                     return false;
                     // beerocks_agent will not receive ACTION_BACKHAUL_CONNECTED_NOTIFICATION
                 }
-            }
+            }*/
         }
     }
 
-- 
2.34.1

